add_form.php

```diff
@@ -280,6 +280,14 @@ public function definition() {
             $mform->setDefault('preventsharediptime', $pluginconfig->preventsharediptime);
         }
 
+        // Upload attendance files.
+        $mform->addElement('header', 'headeruploadattendance', get_string('uploadattendance', 'attendance'));
```

Comment:
At the university I work at, lecturers allow their students to scan their student ids (take their own attendance) up to 30 mins before the start of class. With the current plugin, that is not possible for students to perform. Students could only take their attendance when the session has started. 

This setting is meant to cater to these early students by allowing the lecturer to set how early the earliest permissible scan can occur. I chose to make it a setting at the session creating level because here the lecturer only has to set this "earliest time" once for the course if he/she decides to change it from the default 30 mins. I believe this is a necessary option, since many of our lecturers have expressed that they would like the attendance taking process to occur as quickly as possible before class starts so that it does not become a distraction to students.

Reply:
Right - so that relates to issue #297 and has nothing to do with the import feature so should be submitted as a separate pull request/improvement. It's also not implemented quite right so can you please remove it from this patch and we can then deal with it later as a separate improvement? thanks!

Reply:
I understand. It will be removed.